glyph/src/font.rs

```diff
@@ -177,6 +177,10 @@ pub trait Font {
     /// To get the largest image use `u16::MAX`.
     fn glyph_raster_image2(&self, id: GlyphId, pixel_size: u16) -> Option<v2::GlyphImage>;
 
+    fn as_slice(&self) -> &[u8];
+
+    fn into_vec(&self) -> Vec<u8>;
```

Comment:
Adding these are a breaking change I'd rather avoid, see https://github.com/alexheretic/ab-glyph/issues/97.

Instead we can could a new trait for these, which avoids requiring these be implemented by external `Font` impls.

Reply:
Does that solve the problem though? Unless im misunderstanding things to get this passed through to FontArc you'd end up with something like:

````
pub trait FontDate {
  pub fn as_slice(&self) -> &[u8];
}

pub trait ExtendedFont: Font + FontData;

pub struct FontArc(Arc<dyn ExtendedFont + Send + Sync + 'static>);
````

So its a breaking change to `FontArc` just not `Font`.

Reply:
One way could be to add a default panicking impl to `Font` & implement it in FontArc, FontRef, FontVec. Also maybe this method should be called `font_data` since it is calling `raw_face().data`.

```rust
pub trait Font {
    ...
    // TODO remove default impl on next breaking change
    fn font_data(&self) -> &[u8] {
        // panic impl prevents this method from breaking external Font impls
        unimplemented!()
    }
}
```

Reply:
Done it that way and pushed.

Main difference here is changing the macro slightly to allow FontRef and FontVec to use different implementations instead of getting the raw face for FontVec when not needed.